### PR TITLE
Single execute no time advance

### PR DESCRIPTION
**Pull request by delandmeterp**
The kernel will be executed only once, with dt=0.

This should then not be used for advection.

Works with both jit and scipy.

### CODE REVIEW

File: parcels/particleset.py [hunk 1]
```diff
@@ -273,20 +273,27 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
...
                 logger.warning("Negating dt because running in time-backward mode")
             if interval > 0.:
                 interval *= -1.
                 logger.warning("Negating interval because running in time-backward mode")
 
+        if np.allclose(endtime, starttime) or interval == 0 or dt == 0 or runtime == 0:
+            timeleaps = 1
+            dt = 0
+            runtime = 0
+            endtime = starttime
```
Comment (erikvansebille) [inline parcels/particleset.py]: Add a `logger.warning_once` here that kernels are executed without incrementing time?

File: tests/test_particle_sets.py [hunk 2]
```diff
@@ -277,3 +277,23 @@ def test_pfile_array_remove_all_particles(fieldset, mode, tmpdir, pfile_type, np
...
+    if test == 'dt':
+        pset.execute(pset.Kernel(SetLat), starttime=0., endtime=1., dt=0.)
+    elif test == 'endtime':
+        pset.execute(pset.Kernel(SetLat), starttime=0., endtime=0., dt=1.)
+    assert np.allclose(pset.particles[0].lat, .6)
+    assert np.allclose(pset.particles[1].lat, .6)
+    assert np.allclose(pset.particles[0].lon, 0)
+    assert np.allclose(pset.particles[1].lon, 1)
+    assert np.allclose(pset.particles[1].time, 0)
+    assert np.allclose(pset.particles[0].time, 0)
```
Comment (erikvansebille) [inline tests/test_particle_sets.py]: Or define first `lon = np.linspace(0, 1, npart, dtype=np.float32)` and then `pset = ParticleSet(..., lon=lon, ...)`

Then, the asserts could become
```
assert np.allclose([p.lon for p in pset], 0.6)
assert np.allclose([p.lon for p in pset], lon)
```
(untested)

File: tests/test_particle_sets.py [hunk 3]
```diff
@@ -277,3 +277,23 @@ def test_pfile_array_remove_all_particles(fieldset, mode, tmpdir, pfile_type, np
...
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('test', ['dt', 'endtime'])
+def test_pset_execute_dt_0(fieldset, mode, test, npart=2):
+    def SetLat(particle, fieldset, time, dt):
+        particle.lat = .6
+    pset = ParticleSet(fieldset, pclass=ptype[mode],
+                       lon=np.linspace(0, 1, npart, dtype=np.float32),
+                       lat=np.linspace(1, 0, npart, dtype=np.float32))
+    if test == 'dt':
```
Comment (erikvansebille) [inline tests/test_particle_sets.py]: If-statement can be removed by passing values for `dt` and `endtime` in pairs in the `@pytest` call. E.g.
```
@pytest.mark.parametrize('endtime', 'dt', [(1., 0.), (0., 1.)])
```
(untested)

File: parcels/particleset.py (hunk 1 above)
Comment (delandmeterp) [reply parcels/particleset.py]: It is done, but git didn't see it (it's just next line)

**Pull request merged**